pinot-core/src/main/java/org/apache/pinot/core/accounting/PerQueryCPUMemAccountantFactory.java

```diff
@@ -132,6 +132,8 @@ public static class PerQueryCPUMemResourceUsageAccountant implements ThreadResou
 
     protected final Set<String> _inactiveQuery;
 
+    protected Set<String> _cancelSentQueries;
```

Comment:
In the incident we encountered recently, we saw this framework attempting to kill the same query (which had a 60 second timeout) for over 3 hours - is the inactive query set not being used at all? Could we leverage that here or is it intentional to split out this cancel sent queries set wherein we might kill multiple queries consecutively?

Reply:
Makes sense, thanks for the explanation.